components/brave_ads/core/internal/ads/serving/choose/sample_ads.h

```diff
@@ -53,6 +66,19 @@ absl::optional<T> SampleAdFromPredictors(
   return absl::nullopt;
 }
 
+inline std::vector<double> ComputeProbabilities(std::vector<int> scores) {
+  std::vector<double> probabilities;
+
+  const double normalizing_constant = CalculateNormalizingConstantBase(scores);
+
+  probabilities.reserve(scores.size());
```

Comment:
`std::vector<double> probabilities(scores.size())` and then you can remove `.reserve`